consent/types.go

```diff
@@ -53,18 +58,14 @@ type RequestDeniedError struct {
 }
 
 func (e *RequestDeniedError) toRFCError() *fosite.RFC6749Error {
-	if e.Name == "" {
-		e.Name = fosite.ErrInvalidRequest.Name
+	if e.Name == "" && e.Description == "" && e.Hint == "" {
+		e.Name = requestDeniedErrorName
+		e.Description = requestDeniedErrorDescription
 	}
+
 	if e.Code == 0 {
 		e.Code = fosite.ErrInvalidRequest.Code
 	}
-	if e.Description == "" {
-		e.Description = fosite.ErrInvalidRequest.Description
-	}
-	if e.Hint == "" {
-		e.Hint = fosite.ErrInvalidRequest.Hint
-	}
 	if e.Debug == "" {
 		e.Debug = fosite.ErrInvalidRequest.Debug
```

Comment:
Probably makes sense to remove this too?

Reply:
Yeah I think so.

Should the default error code stay as `fosite.ErrInvalidRequest` (400)?